### DISCUSSION TITLE
What is the correct if condition syntax for checking matrix.os version?

### DISCUSSION THREAD

**Discussion by Siedlerchr** [community/community#25554] [Actions]
<p>Hi,</p>
<p>I previously used travis with a build matrix and could check in if condtions for the os-version/type, however in GithubActions I couldn’t figure out what the correct syntax is:</p>
<p>I tested both with and without expression braces:</p>
<pre><code>- if: matrix.os == 'ubuntu-latest'
      run: |
        ionic cordova platform add android
        ionic cordova build android
</code></pre>
<p>and something like:</p>
<pre><code>- if: ${{ matrix.os == 'macOS-10.14'}}
run: | ionic cordova platform add ios
              ionic cordova build ios
</code></pre>
<p>Both give me errors:</p>
<pre><code>- Your workflow file was invalid: .github/workflows/main.yml (Line: 33, Col: 11): Unrecognized named-value: 'matrix'. Located at position 1 within expression: matrix.os == 'macOS-10.14'
</code></pre>

**Comment by Siedlerchr** [comment, marked as answer]
<p>I figured it out and now it works:</p>
<pre><code>- if: matrix.os == 'macOS-10.14'
      run: |
        ionic cordova platform add ios
        ionic cordova build ios

 - if: matrix.os == 'ubuntu-latest'
      run: |
        ionic cordova platform add android
        ionic cordova build android
</code></pre>

**Comment by nigelhorne** [comment]
<p>Does’t work.  I get “You have an error in your yaml syntax on line 30” with this:</p>
<p>      - name: Install Depencies unless Windows</p>
<p>        - if: matrix.os != ‘windows-latest’</p>
<p>          run: cpanm -iqn BSD::Resource</p>
<p>Of course, it’s not helpful that Github doesn’t tell you what the syntax error is, or the place in the line that the error occured.</p>

**Reply by zekrom-vale** [reply]
If the yaml syntax is malformed try a linter like https://www.yamllint.com/

**Comment by Siedlerchr** [comment]
<pre><code>- name: run on mac
      if: startsWith(matrix.os, 'macOS')
      run: |
        ionic cordova platform add ios
        ionic cordova build ios
        mkdir -pv output
    - name: run on linux
      if: startsWith(matrix.os,'ubuntu')
      run: |
        npm run test-headless
        ionic cordova platform add android
        ionic cordova build android
        mkdir -pv output
        readlink -f output
</code></pre>
<p>This works fine for me. Maybe your indentation is wrong?</p>

**Comment by nigelhorne** [comment]
<p>I checked and double checked the indentation.  Looks right to me.  And there are no tabs either.</p>

**Comment by nigelhorne** [comment]
<p>Looking more, the indentation isn’t clear to me.  In your answer the ‘run’ is under the ‘if’; whereas in the ‘solved’ response it’s indented by four spaces.  I indented by two.</p>

**Comment by nigelhorne** [comment]
<p>That was it.  It wants no indend, the “solved” solution is wrong.</p>

**Comment by Siedlerchr** [comment]
<p>I think it could be a copy paste problem with the forum here which killed the right indentations</p>

**Comment by zekrom-vale** [comment]
Having separate names makes it work fine
```yaml
    - name: Setup wine Ubuntu
      if: matrix.os == 'ubuntu-latest'
      run: |
        sudo apt install wine
    - name: Setup wine macOS
      if: matrix.os =='macos-latest'
      run: |
        brew install wine-stable --cask
```

This is way better than the shell way, as windows breaks when it tries this command:
```shell
if [ "$RUNNER_OS" == "Linux" ]; then
  sudo apt install wine
elif [ "$RUNNER_OS" == "macOS" ]; then
   brew install wine-stable --cask
fi
```
